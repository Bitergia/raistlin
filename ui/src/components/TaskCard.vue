<template>
  <v-card :to="{path: '/tasks/' + task.task_id }">
    <div class="d-flex flex-no-wrap justify-space-between pa-4">
      <div class="flex-grow-1">
        <v-card-title>
          {{ task.task_id }}
          <v-chip
            size="small"
            class="ml-2 text-capitalize"
            :color="task.status.toLowerCase()"
          >
            {{ task.status }}
          </v-chip>
        </v-card-title>
        <v-card-text>
          <v-row>
            <v-col cols="6">
              <p class="mb-3">
                <span class="font-weight-medium">{{ task.jobs.length }}</span> jobs
              </p>
              <v-icon
                v-for="job in task.jobs.slice(0,10)"
                :key="job.job_number"
                :color="job.job_status"
                icon="mdi-square"
              />
            </v-col>
            <v-col>
              <p class="mb-3">
                <v-icon>
                  {{ 'mdi-' + task.backend }}
                </v-icon>
                {{ task.backend }}
              </p>
              <p>
                <v-icon icon="mdi-calendar-month" />
                {{ prettyDate(task.created_on) }}
              </p>
            </v-col>
          </v-row>
        </v-card-text>
      </div>
      <v-card-actions>
        <v-btn
          icon="mdi-delete"
          color="amber-darken-4"
          @click.prevent="$emit('openModal')"
        />
      </v-card-actions>
    </div>
  </v-card>
</template>

<script>
import { prettyDate } from './../utils/filters'

export default {
  name: 'TaskCard',
  props: {
    task: {
      type: Object,
      required: true
    }
  },
  emits: ['openModal'],
  setup() {
    return { prettyDate }
  }
}
</script>
