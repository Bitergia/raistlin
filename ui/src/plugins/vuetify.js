import 'vuetify/styles';
import { createVuetify } from 'vuetify';
import * as components from 'vuetify/components';
import * as directives from 'vuetify/directives';
import '@mdi/font/css/materialdesignicons.css';

export default createVuetify({
  components,
  directives,
  icons: {
    defaultSet: 'mdi'
  },
  theme: {
    themes: {
      light: {
        colors: {
          completed: '#4CAF50',
          failed: '#B71C1C',
          enqueued: '#9E9E9E',
          running: '#ffd21f',
          scheduled: '#45b6f7',
          started: '#ffd21f',
          queued: '#9E9E9E',
          finished: '#4CAF50'
        }
      }
    }
  }
});
