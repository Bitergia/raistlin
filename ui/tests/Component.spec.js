import { mount } from '@vue/test-utils'
import TaskCard from './../src/components/TaskCard.vue'
import vuetify from './../src/plugins/vuetify'

describe("Vue + Vuetify component testing", () => {
  const mountFunction = (options) => {
    return mount(TaskCard, {
      global: {
        plugins: [vuetify],
        stubs: ["router-link"]
      },
      props: {
        task: {
          task_id: "arthur.git",
          status: "scheduled",
          backend: "git",
          created_on: 1675682110.769566,
          jobs: [
            {
              job_number: 1,
              job_status: "finished"
            }
          ]
        }
      },
      ...options,
    });
  };
  test('mount component', async () => {
    const wrapper = mountFunction()

    expect(wrapper.text()).toContain('arthur.git')
  })


  test('trigger event on Vuetify component', async () => {
    const wrapper = await mountFunction()
    wrapper.find('button').trigger('click')

    expect(wrapper.emitted().openModal).toBeTruthy()
  })

  test('check snapshots', async () => {
    const wrapper = mountFunction()

    expect(wrapper.html()).toMatchSnapshot()
  })
})
