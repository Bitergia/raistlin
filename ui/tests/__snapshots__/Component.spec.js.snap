// Vitest Snapshot v1

exports[`Vue + Vuetify component testing > check snapshots 1`] = `
"<a class=\\"v-card v-card--link v-theme--light v-card--density-default v-card--variant-elevated\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
      <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
        <div class=\\"v-progress-linear__indeterminate\\">
          <div class=\\"v-progress-linear__indeterminate long\\"></div>
          <div class=\\"v-progress-linear__indeterminate short\\"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class=\\"d-flex flex-no-wrap justify-space-between pa-4\\">
    <div class=\\"flex-grow-1\\">
      <div class=\\"v-card-title\\">arthur.git <span class=\\"v-chip v-theme--light text-scheduled v-chip--density-default v-chip--size-small v-chip--variant-tonal ml-2 text-capitalize\\" draggable=\\"false\\"><!----><span class=\\"v-chip__underlay\\"></span>
        <!---->
        <!---->scheduled
        <!---->
        <!----></span>
      </div>
      <div class=\\"v-card-text\\">
        <div class=\\"v-row\\">
          <div class=\\"v-col v-col-6\\">
            <p class=\\"mb-3\\"><span class=\\"font-weight-medium\\">1</span> jobs </p><i class=\\"mdi-square mdi v-icon notranslate v-theme--light v-icon--size-default text-finished\\" aria-hidden=\\"true\\"></i>
          </div>
          <div class=\\"v-col\\">
            <p class=\\"mb-3\\"><i class=\\"mdi-git mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i> git</p>
            <p><i class=\\"mdi-calendar-month mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i> 2023/2/6 11:15:10</p>
          </div>
        </div>
      </div>
    </div>
    <div class=\\"v-card-actions\\"><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light text-amber-darken-4 v-btn--density-default v-btn--size-default v-btn--variant-text\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
        <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-delete mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
        <!---->
        <!---->
      </button></div>
  </div>
  <!----><span class=\\"v-card__overlay\\"></span><span class=\\"v-card__underlay\\"></span>
</a>"
`;
